program/src/instructions/legacy/buy.rs

```diff
@@ -209,16 +208,19 @@ pub fn process_buy_legacy<'info, 'b>(
     } = calc_fees(CalcFeesArgs {
         amount,
         tnsr_discount: false,
-        total_fee_bps: TCOMP_FEE_BPS,
+        total_fee_bps: TAKER_FEE_BPS,
         broker_fee_pct: BROKER_FEE_PCT,
         maker_broker_pct: MAKER_BROKER_PCT,
     })?;
 
     let creator_fee = calc_creators_fee(
         metadata.seller_fee_basis_points,
         amount,
-        metadata.token_standard,
-        optional_royalty_pct,
+        if is_royalty_enforced(metadata.token_standard) {
```

Comment:
arent we doing this inside of the fn?

Reply:
Now that we support so many different standards it doesn't make sense to have this logic specific to legacy/pNFT Metaplex NFTs in the generic `calc_creators_Fee` function, so Richard moved it out into a separate helper.